PyPDF2/_merger.py

```diff
@@ -285,7 +312,7 @@ def write(self, fileobj: StrByteType) -> None:
         self._write_outline()
 
         # Write the output to the file
-        self.output.write(fileobj)
+        my_file, fileobj = self.output.write(fileobj)
 
         if my_file:
             fileobj.close()
```

Comment:
My preference would be to avoid overwriting the `fileobj` passed into the surrounding `write` function with got returned from `self.output.write`:
```suggestion
        my_file, ret_fileobj = self.output.write(fileobj)

        if my_file:
            ret_fileobj.close()
```

It makes it more clear on what's getting operated on imo.